src/test/compile-fail/coherence-inherited-assoc-ty-cycle-err.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Formerly this ICEd with the following message:
+// Tried to project an inherited associated type during coherence checking,
+// which is currently not supported.
```

Comment:
It here: https://github.com/rust-lang/rust/blob/4d09a0eb5d201c831d19f3fe93fbe636eca7d97d/src/librustc/traits/project.rs#L989

But it has this nice, long comment which made me reluctant to remove it. I could move the comment to `assoc_ty_def()`, make its return value non-optional, and then remove the branch though.